pkg/server/io/container_io.go

```diff
@@ -105,18 +105,20 @@ func (c *ContainerIO) Config() cio.Config {
 // to output stream.
 func (c *ContainerIO) Pipe() {
 	wg := c.closer.wg
-	wg.Add(1)
-	go func() {
-		if _, err := io.Copy(c.stdoutGroup, c.stdout); err != nil {
-			logrus.WithError(err).Errorf("Failed to pipe stdout of container %q", c.id)
-		}
-		c.stdout.Close()
-		c.stdoutGroup.Close()
-		wg.Done()
-		logrus.Infof("Finish piping stdout of container %q", c.id)
-	}()
+	if c.stdout != nil {
+		wg.Add(1)
+		go func() {
+			if _, err := io.Copy(c.stdoutGroup, c.stdout); err != nil {
+				logrus.WithError(err).Errorf("Failed to pipe stdout of container %q", c.id)
+			}
+			c.stdout.Close()
```

Comment:
are we only worried about null stdout/err at the start.. or could be be nulled out after the create?

Reply:
It could be nil if the task status is unknown. That's why we handle it here.

See https://github.com/containerd/containerd/pull/3611

Reply:
maybe I'm reading the code wrong.. but it reads to me like if stdout isn't set then closer isn't set?

Reply:
Oops, I think I missed a fix.

Reply:
Added the missing fix... Confused by my PRs. :P